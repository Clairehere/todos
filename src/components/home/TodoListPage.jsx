import {withTranslation} from "react-i18next"
import {connect} from 'react-redux'
import withStyles from "@material-ui/core/styles/withStyles"
import React, {useState} from "react"
import {compose} from 'recompose'
import PropTypes from 'prop-types'
import TaskList from "../tasks/TaskList"
import NewTaskDialog from "../tasks/NewTaskDialog"
import Button from "@material-ui/core/Button"
import AddIcon from '@material-ui/icons/Add'

const styles = () => ({
    title:{
      marginLeft:'10%'
    },
    container:{
      marginLeft:'30%'
    }
})


export const TodoListPage = ({t, classes, lists, selectedList}) => {
  const [dialogOpen, setDialogOpen] = useState(false)

  return (
    <>
     { selectedList &&
      <div className={classes.container}>
        <h2> {selectedList.name } </h2>
        <Button aria-label="archive" style={{ marginTop: '5px' }}  onClick={() => setDialogOpen(true)}>
          <AddIcon edge={"start"}/> {t('task.add')}</Button>
        <TaskList/>
      </div>}
      <NewTaskDialog open={dialogOpen} handleClose={() => setDialogOpen(false)} t={t} />
    </>
  )
}

const mapStateToProps = (state) => ({
    lists: state.data.initialLists || [],
    selectedList: state.data.selectedList,
  })


TodoListPage.propTypes = {
  t: PropTypes.func.isRequired,
  lists: PropTypes.array.isRequired,
  classes: PropTypes.object.isRequired,
}

export default compose(
  withStyles(styles),
  withTranslation('common'),
  connect(mapStateToProps),
)(TodoListPage)