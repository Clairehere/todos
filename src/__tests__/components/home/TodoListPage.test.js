import TodoListPage from "../../../components/home/TodoListPage"
import React from "react"
import {shallow, configure} from "enzyme"

const mockStore = configureMockStore()
const store = mockStore({})
import {Provider} from "react-redux"
import configureMockStore from "redux-mock-store"
import Adapter from 'enzyme-adapter-react-16'
import {checkProps} from "../../functions"
import {array, func, object, lists} from "../../fixtures"


configure({adapter: new Adapter()});

describe(">>> TodoListPage --- Snapshot", () => {
  it("+++capturing Snapshot of TodoListPage - with list", () => {
    expect(
      shallow(
        <Provider store={store}>
          <TodoListPage
            t={t => t}
            lists={[]}
            classes={{}}
            selectedList={lists[0]}
            editList={() => null}/>
        </Provider>
      )).toMatchSnapshot()
  })
  it("+++capturing Snapshot of TodoListPage - no list", () => {
    expect(
      shallow(
        <Provider store={store}>
          <TodoListPage
            t={t => t}
            lists={[]}
            classes={{}}
            selectedList={{}}
            editList={() => null}/>
        </Provider>
      )).toMatchSnapshot()
  })

})


describe('>>> M O R E   A C T I O N   I T E M --- Shallow render component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <TodoListPage
          t={t => t}
          lists={[]}
          classes={{}}
          selectedList={lists[0]}
          editList={() => null}/>
      </Provider>)
  })

  it('+++ render the DUMB component', () => {
    expect(wrapper.length).toEqual(1)
  })
})


describe('>>> TodoListPage --- checking PropTypes', () => {
  it('+++ should not throw warning', () => {
    const expectedProps = {
      t: func,
      lists: array,
      classes: object,
      selectedList: object,
      editList: func,
    }

    const propsErr = checkProps(TodoListPage, expectedProps)
    expect(propsErr).toEqual(undefined)
  })
})

