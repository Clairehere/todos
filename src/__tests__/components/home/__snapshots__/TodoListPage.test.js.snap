// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> TodoListPage --- Snapshot +++capturing Snapshot of TodoListPage - no list 1`] = `ShallowWrapper {}`;

exports[`>>> TodoListPage --- Snapshot +++capturing Snapshot of TodoListPage - with list 1`] = `ShallowWrapper {}`;

